internal/tools/k8s/helpers_test.go

```diff
@@ -26,10 +28,18 @@ func TestWaitForPodRunning(t *testing.T) {
 		assert.Equal(t, corev1.PodRunning, pod.Status.Phase)
 	})
 	t.Run("wait for running", func(t *testing.T) {
-		pod, err := testClient.WaitForPodRunning(namespace, podName, 30)
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+		pod, err := testClient.WaitForPodRunning(namespace, podName, ctx)
 		require.NoError(t, err)
 		assert.Equal(t, corev1.PodRunning, pod.Status.Phase)
 	})
+	t.Run("don't wait for running", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 0*time.Second)
```

Comment:
I wonder if constructing with `0` as the timeout guarantees the `context` will be immediately done or if a fast test might sometimes "pass". Perhaps use `context.WithCancel` and then immediately cancel it instead?